lib/services/services.c

```diff
@@ -1079,3 +1080,82 @@ resources_list_agents(const char *standard, const char *provider)
 
     return NULL;
 }
+
+gboolean
+resources_valid_tuple(const char *standard, const char *provider, const char *agent)
+{
+    GList *standards = NULL;
+    GList *providers = NULL;
+    GList *agents = NULL;
+    GListPtr iter = NULL;
+    gboolean rc = FALSE;
+
+    if (crm_str_eq(standard, "stonith", TRUE)) {
+        stonith_t *st = NULL;
+        stonith_key_value_t *devices = NULL;
+        stonith_key_value_t *dIter = NULL;
+        int timeout = 120;
```

Comment:
It's probably worth passing timeout as an argument, and using this default if it's 0